src/runtime/execution.rs

```diff
@@ -644,14 +647,14 @@ impl ExecutionState {
         self.context_switches += 1;
 
         match self.config.max_steps {
-            MaxSteps::FailAfter(n) if self.current_schedule.len() >= n => {
+            MaxSteps::FailAfter(n) if self.current_schedule.len() - self.steps_reset_at >= n => {
```

Comment:
For readability, would be nice to extract the calculation (which also happens below) this into a function.

Reply:
Ack done